### ISSUE TITLE
t function with plurals returns type never after migrating to v21.x.x

### ISSUE THREAD

**Issue by wolkyura** [i18next/react-i18next#1451]
## 🐛 Bug Report

After upgrading `i18next` to latest version from `20.x.x` version, `t('ns:key', { count: 1 })` function with count option always returns value with type `never` (on version `20.x.x` - `string` as expected)

<img width="400" alt="Screen Shot 2022-02-16 at 3 32 37 PM" src="https://user-images.githubusercontent.com/54850073/154286153-ac5b7da9-6db4-4a35-9123-7c167f3dd3e9.png">

## To Reproduce

```json
{
  "key_one": "value",
  "key_other": "values"
}
```

```ts
const { t } = useTranslation(['common']);
const value = t('common:key', { count: 10 }); // type of value - `never`
```
`react-18next.d.ts`:

```ts
import { resources } from './i18n';

declare module 'react-i18next' {
  interface CustomTypeOptions {
    resources: typeof resources['en'];
  }
}
```

`i18n.ts`:

```ts
import i18n from 'i18next';
import detector from 'i18next-browser-languagedetector';
import { initReactI18next } from 'react-i18next';

import { en } from './locales/en';

export const resources = {
  en,
} as const;

i18n
  .use(detector)
  .use(initReactI18next)
  .init({
    interpolation: {
      escapeValue: false,
    },
    lng: 'en',
    resources,
  });

export default i18n;

```

## Expected behavior

I would expect `string` type of returning value `t('common:key', { count: 10 })`

```ts
const { t } = useTranslation(['common']);
const value = t('common:key', { count: 10 }); // type of value - `string`
```

## Your Environment

- *runtime version*: node v14
- *i18next version*: 21.6.6
- *react-i18next*: 11.15.3,
- *os*: Mac

**Comment by pedrodurek** [comment]
Hey @wolkyura, that's definitely a bug, I'll try to fix it today

**Comment by adrai** [comment]
fixed with v11.15.5
